### ISSUE TITLE
IndexError: list index out of range in output_pocket_pdb (segment_pockets.py)

### ISSUE THREAD

**Issue by pjmartel** [devalab/DeepPocket#3]
Hello, 

I'm trying to run the the Predicting Binding Site section example:

`python predict.py -p protein.pdb -c first_model_fold1_best_test_auc_85001.pth.tar -s seg0_best_test_IOU_91.pth.tar -r 3
`

But it crashes with the following errors:

`***** POCKET HUNTING BEGINS ***** 
***** POCKET HUNTING ENDS ***** 
/usr/local/lib/python3.7/dist-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 1951.
  PDBConstructionWarning,
/usr/local/lib/python3.7/dist-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 2008.
  PDBConstructionWarning,
/content/DeepPocket/rank_pockets.py:87: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  all_probs.append(F.softmax(output).detach().cpu())
@> 1674 atoms and 1 coordinate set(s) were parsed in 0.01s.
Traceback (most recent call last):
  File "predict.py", line 116, in <module>
    test(seg_model, seg_eptest, seg_gmaker,device,dx_name, args)
  File "/content/DeepPocket/segment_pockets.py", line 142, in test
    output_pocket_pdb(dx_name+'_pocket'+str(count)+'.pdb',prot_prody,pred_aa)
  File "/content/DeepPocket/segment_pockets.py", line 82, in output_pocket_pdb
    pocket=prot_prody.select(sel_str)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/atomic.py", line 232, in select
    return SELECT.select(self, selstr, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 885, in select
    indices = self.getIndices(atoms, selstr, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 943, in getIndices
    torf = self.getBoolArray(atoms, selstr, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 995, in getBoolArray
    tokens = parser(selstr, parseAll=True)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 1100, in _noParser
    return [self._default(selstr, 0, selstr.split())]
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 1118, in _default
    torf, err = self._and2(sel, loc, tokens)
  File "/usr/local/lib/python3.7/dist-packages/prody/atomic/select.py", line 1319, in _and2
    firsttoken = tokens[0] if not isinstance(tokens[0], Iterable) else list(tokens[0])
IndexError: list index out of range`

Ubuntu 18.04 on Google Colab (pytorch 1.9+cuda 10.2) with prody version 2.0.

**Comment by RishalAggarwal** [comment]
Hi, could you also share the pdb file to reproduce the error

**Comment by pjmartel** [comment]
My apologies. The pdb file is `1k2c.pdb` -  in fact I used the following  command line:

`python predict.py -p 1k2c.pdb -c first_model_fold1_best_test_auc_85001.pth.tar -s seg0_best_test_IOU_91.pth.tar -r 3.`

**Comment by pjmartel** [comment]
Ok, I tried to run it again in Colab (factory resetting the runtime and installing everything again), and now I get a different error:
`***** POCKET HUNTING BEGINS ***** 
***** POCKET HUNTING ENDS ***** 
mkdir: cannot create directory ‘1k2c_nowat_out/pockets’: File exists
/usr/local/lib/python3.7/dist-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 1951.
  PDBConstructionWarning,
/usr/local/lib/python3.7/dist-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 2008.
  PDBConstructionWarning,
/content/DeepPocket/rank_pockets.py:87: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  all_probs.append(F.softmax(output).detach().cpu())
Traceback (most recent call last):
  File "predict.py", line 115, in <module>
    seg_model, seg_gmaker, seg_eptest=get_model_gmaker_eprovider(seg_types,1,seg_model,seg_checkpoint,dims=32)
  File "predict.py", line 62, in get_model_gmaker_eprovider
    model.load_state_dict(checkpoint['model_state_dict'])
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1407, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for DataParallel:
	Missing key(s) in state_dict: "module.in1.block.0.weight", "module.in1.block.0.bias", "module.in1.block.1.weight", "module.in1.block.1.bias", "module.in1.block.1.running_mean", "module.in1.block.1.running_var", "module.in1.block.3.weight", "module.in1.block.3.bias", "module.in1.block.4.weight", "module.in1.block.4.bias", "module.in1.block.4.running_mean", "module.in1.block.4.running_var", "module.down1.block.1.block.0.weight", "module.down1.block.1.block.0.bias", "module.down1.block.1.block.1.weight", "module.down1.block.1.block.1.bias", "module.down1.block.1.block.1.running_mean", "module.down1.block.1.block.1.running_var", "module.down1.block.1.block.3.weight", "module.down1.block.1.block.3.bias", "module.down1.block.1.block.4.weight", "module.down1.block.1.block.4.bias", "module.down1.block.1.block.4.running_mean", "module.down1.block.1.block.4.running_var", "module.down2.block.1.block.0.weight", "module.down2.block.1.block.0.bias", "module.down2.block.1.block.1.weight", "module.down2.block.1.block.1.bias", "module.down2.block.1.block.1.running_mean", "module.down2.block.1.block.1.running_var", "module.down2.block.1.block.3.weight", "module.down2.block.1.block.3.bias", "module.down2.block.1.block.4.weight", "module.down2.block.1.block.4.bias", "module.down2.block.1.block.4.running_mean", "module.down2.block.1.block.4.running_var", "module.down3.block.1.block.0.weight", "module.down3.block.1.block.0.bias", "module.down3.block.1.block.1.weight", "module.down3.block.1.block.1.bias", "module.down3.block.1.block.1.running_mean", "module.down3.block.1.block.1.running_var", "module.down3.block.1.block.3.weight", "module.down3.block.1.block.3.bias", "module.down3.block.1.block.4.weight", "module.down3.block.1.block.4.bias", "module.down3.block.1.block.4.running_mean", "module.down3.block.1.block.4.running_var", "module.down4.block.1.block.0.weight", "module.down4.block.1.block.0.bias", "module.down4.block.1.block.1.weight", "module.down4.block.1.block.1.bias", "module.down4.block.1.block.1.running_mean", "module.down4.block.1.block.1.running_var", "module.down4.block.1.block.3.weight", "module.down4.block.1.block.3.bias", "module.down4.block.1.block.4.weight", "module.down4.block.1.block.4.bias", "module.down4.block.1.block.4.running_mean", "module.down4.block.1.block.4.running_var", "module.up1.up_s.weight", "module.up1.up_s.bias", "module.up1.convT.block.0.weight", "module.up1.convT.block.0.bias", "module.up1.convT.block.1.weight", "module.up1.convT.block.1.bias", "module.up1.convT.block.1.running_mean", "module.up1.convT.block.1.running_var", "module.up1.convT.block.3.weight", "module.up1.convT.block.3.bias", "module.up1.convT.block.4.weight", "module.up1.convT.block.4.bias", "module.up1.convT.block.4.running_mean", "module.up1.convT.block.4.running_var", "module.up2.up_s.weight", "module.up2.up_s.bias", "module.up2.convT.block.0.weight", "module.up2.convT.block.0.bias", "module.up2.convT.block.1.weight", "module.up2.convT.block.1.bias", "module.up2.convT.block.1.running_mean", "module.up2.convT.block.1.running_var", "module.up2.convT.block.3.weight", "module.up2.convT.block.3.bias", "module.up2.convT.block.4.weight", "module.up2.convT.block.4.bias", "module.up2.convT.block.4.running_mean", "module.up2.convT.block.4.running_var", "module.up3.up_s.weight", "module.up3.up_s.bias", "module.up3.convT.block.0.weight", "module.up3.convT.block.0.bias", "module.up3.convT.block.1.weight", "module.up3.convT.block.1.bias", "module.up3.convT.block.1.running_mean", "module.up3.convT.block.1.running_var", "module.up3.convT.block.3.weight", "module.up3.convT.block.3.bias", "module.up3.convT.block.4.weight", "module.up3.convT.block.4.bias", "module.up3.convT.block.4.running_mean", "module.up3.convT.block.4.running_var", "module.up4.up_s.weight", "module.up4.up_s.bias", "module.up4.convT.block.0.weight", "module.up4.convT.block.0.bias", "module.up4.convT.block.1.weight", "module.up4.convT.block.1.bias", "module.up4.convT.block.1.running_mean", "module.up4.convT.block.1.running_var", "module.up4.convT.block.3.weight", "module.up4.convT.block.3.bias", "module.up4.convT.block.4.weight", "module.up4.convT.block.4.bias", "module.up4.convT.block.4.running_mean", "module.up4.convT.block.4.running_var", "module.conv.weight", "module.conv.bias". 
	Unexpected key(s) in state_dict: "unit0_conv.weight", "unit0_conv.bias", "unit1_conv.weight", "unit1_conv.bias", "unit2_conv.weight", "unit2_conv.bias", "unit3_conv.weight", "unit3_conv.bias", "unit4_conv.weight", "unit4_conv.bias", "unit5_conv.weight", "unit5_conv.bias", "final_fc.weight", "final_fc.bias". `

Now this is really strange, because I used the exact same weigth file as before... and all software versions and other input files are the same!

**Comment by RishalAggarwal** [comment]
Thank you for bringing this to my attention, the script could not handle cases where no binding site residues are predicted (which unfortunately can be the case sometimes with Deep Learning models). Fixed the issue it should work now. Also, now I've made sure that the script doesn't run if the model weights are not specified, hopefully that addresses the other problem you had. Closing the issue now, do reopen if you have any more issues.
